graphql/schema.js

```diff
@@ -156,8 +158,42 @@ const courseOfferingType = new GraphQLObjectType({
   fields: () => ({
     year: { type: GraphQLString },
     quarter: { type: GraphQLString },
+    instructors: { 
+      type: GraphQLList(instructorType),
+      resolve: (temp) => {
+        return temp.instructors.map((name) => {
+
+          //Fetch all possible ucinetids from the instructor.
+          let ucinetids = getUCINetIDFromName(name);
+          
+          //If only one exists, then we can return the instructor for it.
+          if (ucinetids && ucinetids.length == 1) { return getInstructor(ucinetids[0]); }
+          
+          //If there is more than one, figure it out.
+          else if (ucinetids && ucinetids.length > 1) {
+              
+              //Filter our instructors by those with related departments.
+              let course_dept = getCourse(temp.course).department;
+              let instructors = ucinetids.map(id => getInstructor(id)).filter( temp => temp.related_departments.includes(course_dept));
+              
+              //If only one is left, we can return it.
+              if (instructors.length == 1) {  return getInstructor(instructors[0].ucinetid); } 
+
+              //Filter instructors by those that taught the course before.
+              instructors = instructors.filter( inst => {
+                  return inst.course_history.map((course) => getCourse(course.replace(/ /g, ""))).includes(temp.course);
+              });
```

Comment:
I wonder if we should always filter by those who have taught the course before, to reduce the chances of a false positive?
```suggestion
              //Filter instructors by those that taught the course before.
              instructors = instructors.filter( inst => {
                  return inst.course_history.map((course) => getCourse(course.replace(/ /g, ""))).includes(temp.course);
              });
```

Reply:
This seems like a tradeoff of should it be more likely that there is no instructor found or that there is an instructor, but it may be wrong. I think it could be better to have no instructor found at all then be wrong, but what I was thinking is that these instructors already have the exact same first initial and last name, so it is extremely unlikely that they are in the same department and the filter usually finishes there since there is one instructor left. 

In the false positive case, we found an instructor with the same name, and is the only one with that name in that department, but the real instructor is someone else (likely new because it wasn't parsed), or not in the department. I feel that at the point where we find an instructor with that name in that department, and there is only 1, it is that instructor. 

In addition, there are often occasions where an instructor is teaching a new course, and so they would be filtered out when checking if the instructor taught it before, and I feel this would occur much more often than getting that false positive.

Reply:
Ah ok, that makes sense 👍